server/constructors.go

```diff
@@ -9,13 +9,15 @@ import (
 	abci "github.com/tendermint/tendermint/abci/types"
 	dbm "github.com/tendermint/tendermint/libs/db"
 	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/node"
 	tmtypes "github.com/tendermint/tendermint/types"
 )
 
 type (
 	// AppCreator is a function that allows us to lazily initialize an
 	// application using various configurations.
-	AppCreator func(log.Logger, dbm.DB, io.Writer) abci.Application
+	AppCreator func(log.Logger, dbm.DB,
+		io.Writer, node.GenesisDocProvider) abci.Application
```

Comment:
@jaekwon this is my question around design approaches, do you think that it makes sense to include the `GenesisDocProvider` here as a part of the AppCreator? - basically we need to pass back the tendermint genesis in order to retrieve the `MaxBlockGas` which the sdk application needs for this issue